### DISCUSSION TITLE
Can i use existing zerotier-one install with zero-ui

### DISCUSSION THREAD

**Discussion by trivarialthea** [dec0dOS/zero-ui#149] [Q&A]
Hi, i do have an existing install of zerotier-one trough their bash script on a host machine. On the same host machine i have docker. Can i use the zerotier install on the host machine with the zero-ui running in docker?

**Comment by NoobTW** [comment]
Yes, you can do it by editing docker-compose.yml.
1. Comment out the whole "zerotier" service. 
2. change the "ZU_CONTROLLER_ENDPOINT" in zero-ui's environment. Might change to "http://localhost:9993", which depends on your network configuration.
3. Add `network_mode: "host"` in zero-ui service.

**Reply by gt03** [reply]
Hi,

I'm trying to install zero-ui also on a system where zerotier already exist.
I have changed the yaml, but the container don't start up. This is my yaml, can you see whats wrong?

<img width="340" alt="1" src="https://user-images.githubusercontent.com/78665694/236670562-79e3baf4-6aaa-43c3-b04c-dc837178d206.png">

**Comment by AntGod6123** [comment]
For those that saw my earlier posts, I was able to solve the issue. System up and running with the configuration below!

I am trying to do the same, with the docker-compose.yml I am able to access the web gui but no networks show up and I cannot create a new network either. Can anyone please help me in directing to my Ubuntu's Controller? It does not pull data from "/var/lib/zerotier-one/" .I am using a controller on an ubuntu server with a docker zeroUI running. 

version: "3"

services:
  zero-ui:
    image: dec0dos/zero-ui:latest
    container_name: zu-main
    build:
      context: .
      dockerfile: ./docker/zero-ui/Dockerfile
    restart: unless-stopped
    volumes:
      - ./zerotier-one:/var/lib/zerotier-one
      - ./data:/app/backend/data
    environment:
      - ZU_CONTROLLER_ENDPOINT=http://localhost:9993/
      - ZU_SECURE_HEADERS=false
      - ZU_DEFAULT_USERNAME=admin
      - ZU_DEFAULT_PASSWORD=
    network_mode: "host"

**Comment by bruteForce93** [comment]
> For those that saw my earlier posts, I was able to solve the issue. System up and running with the configuration below!
> 
> I am trying to do the same, with the docker-compose.yml I am able to access the web gui but no networks show up and I cannot create a new network either. Can anyone please help me in directing to my Ubuntu's Controller? It does not pull data from "/var/lib/zerotier-one/" .I am using a controller on an ubuntu server with a docker zeroUI running.
> 
> version: "3"
> 
> services: zero-ui: image: dec0dos/zero-ui:latest container_name: zu-main build: context: . dockerfile: ./docker/zero-ui/Dockerfile restart: unless-stopped volumes: - ./zerotier-one:/var/lib/zerotier-one - ./data:/app/backend/data environment: - ZU_CONTROLLER_ENDPOINT=http://localhost:9993/ - ZU_SECURE_HEADERS=false - ZU_DEFAULT_USERNAME=admin - ZU_DEFAULT_PASSWORD= network_mode: "host"

It's not connecting to the controller because the yml is not pointing at it and most definitelly it also cannot reach it.

What I would do is...pull the latest image as per the instructions, cd into the working directory and before composing do the following:

1. First change the volumes to the host controller (the one you already have installed, not the zyclonite one) directory:
```
volumes:
    /var/lib/zerotier-one:var/lib/zerotier-one              # ./zerotier-one:/var/lib/zerotier-one
    - ./data:/app/backend/data
```
2. Sudo nano into /etc/sysctl.conf and uncomment the `#net.ipv4.ip_forward` command.

3. Then enable forwarding: `sudo sysctl -w net.ipv4.ip_forward=1`

4. Sudo nano into /etc/iptables/rules.v4 and insert the following just above command just COMMIT inside the #filter section.

     `-A INPUT -p udp --dport 9993 -j ACCEPT`

5. Delete all ufw rules (if you have created any for zero-ui), reset and disable it and go with iptables instead.

6. Restart the VM.

7. Now add `- ZU_DEFAULT_USERNAME=whateverUsernameYouWantWithoutQuotes` and `- ZU_DEFAULT_PASSWORD=whateverYouWantWithoutQuotes` and then: `docker-compose up -d --no-build` your .yml file.

8. You won't need the local.conf file at all, neither create and setup the controller beforehand.
